hugr-core/src/hugr/validate/test.rs

```diff
@@ -865,11 +864,158 @@ fn test_polymorphic_load() -> Result<(), Box<dyn std::error::Error>> {
     Ok(())
 }
 
+#[test]
+/// Validation errors in a controlflow subgraph.
+fn cfg_children_restrictions() {
```

Comment:
I've moved this out of `extension_tests` as it has nothing to do with extensions